File: PMUG02_16_2021_Project/End_Java/app/src/main/java/com/neudesic/pmugfebappend/database/PersonRepository.java
package com.neudesic.pmugfebappend.database;

import android.app.Application;

import androidx.lifecycle.LiveData;

import com.neudesic.pmugfebappend.model.Person;

import java.util.List;

public class PersonRepository {

    private PersonDao personDao;

    public PersonRepository(Application application) {
        PersonDatabase database = PersonDatabase.getDatabase(application);
        personDao = database.personDao();
    }

    public LiveData<List<Person>> getAllPeople() {
        return personDao.getAllPerson();
    }

    public void insert(Person person) {
        personDao.insert(person);
    }
}

File: PMUG02_16_2021_Project/End_Kotlin/app/src/main/java/com/neudesic/pmug02_16_2021_kotlin/database/PersonRepository.kt
package com.neudesic.pmug02_16_2021_kotlin.database

import android.app.Application
import androidx.lifecycle.LiveData
import com.neudesic.pmug02_16_2021_kotlin.model.Person

class PersonRepository(application: Application) {
    private val personDao: PersonDao
    val allPeople: LiveData<List<Person>>
        get() = personDao.getAllPerson()

    suspend fun insert(person: Person) {
        personDao.insert(person)
    }

    init {
        val database = PersonDatabase.getDatabase(application)
        personDao = database.personDao()
    }
}
